### ISSUE TITLE
Question: running this on Portainer.io?

### ISSUE THREAD

**Issue by Alex000Lexx** [scyto/hdsentinel#4]
Hello!

I've been at this for a few days, trying various configurations on an ASUSTor NAS unit, that is running AMD64. I think the only thing that I am struggling to understand, is how to port your instructions for Synology NAS to something that would run on Portainer.io.

I appreciate you sharing this, and signed up to say thank you! as I would really love to use HD Sentinel on my NAS. 

I think the main challenge is here (in your Synology NAS instructions)-> on volume tab add the following:
hdsentinel folder to mount path /etc/sentinel leave as read/write5.

In Portainer.io I've tried various methods and command line options, and unfortunately, didn't document the one time I did see HD Sentinel run, and then generate an access violation.

my log output looks like this - appreciate any help or guidance you can provide...


111111111111
0000001
100000000000000000000000000000000000000000000000000000000000
000001000000000000000000
01000000000000000000000000000000

111111111111
1111111
100000000010000000001000000000100000000010000000001000000000
111111111111111111111111
11111111111111111111111111111111

111111111111
1111111
crond: USER root pid   9 cmd run-parts /etc/periodic/15min
crond: USER root pid  10 cmd hdsentinel -r /etc/hdsentinel/hdsreport.html -html

Thank you,

Alex

**Comment by scyto** [comment]
Take portainer out of the equation, just incase you are hitting a portainer issue.

You should be able to run this container using the command line shown on home page for this project- the key is to run privileged - if you don't you will see an issue.

Are you using a UI for docker or SSH command line?
If you are using a UI your system must let the docker container run as root.
If your system doesn't run docker as root you may need to override with a UID/GID.

Bottom line try the command line approach first.

**Comment by Alex000Lexx** [comment]
Thanks for the reply scyto! as I am running on an ASUSTor NAS unit, I don't believe it gives me command line as an option - but I will try this on my linux vm to see if I am missing something. 

Re: UI for docker - on the NAS it's Docker-ce and the only UI they expose (that I know of) is Portainer.io. 

I will double check re: running docker container as root, I know that it allows me to use an option to run as privileged, I didn't think of the UID/GID approach, but I will give that a shot. 

Many thanks for replying! I appreciate it - and will go about running this on my linux vm next - just to confirm if it is portainer related.  

I do use linux for my vms, but I find the NAS environment very restrictive, so it might be a lesson learned on that aspect. I'll try there next, just to ensure I can rule portainer.io out of the picture.

Cheers,

Alex

**Comment by scyto** [comment]
@Alex000Lexx how to SSH in https://support.asustor.com/index.php?/Default/Knowledgebase/Article/View/190/22/how-to-ssh-into-my-asustor-nas 
and yeah NAS are less flexible, i feel ya, same on synology.

**Comment by scyto** [comment]
did you create a docker compose file to add the container or deploy a stack for it?
Or did you just click create in the portainer UI.

**Comment by scyto** [comment]
you did set this didn't you?
![image](https://user-images.githubusercontent.com/11369180/77132712-9420e100-6a1d-11ea-8aea-a78791324e22.png)

**Comment by scyto** [comment]
and the host path is some path on your host filesystem that can be written to, this is where the html report is saved.

![image](https://user-images.githubusercontent.com/11369180/77132763-ba468100-6a1d-11ea-8a48-a0987c413d52.png)

**Comment by Alex000Lexx** [comment]
> @Alex000Lexx how to SSH in https://support.asustor.com/index.php?/Default/Knowledgebase/Article/View/190/22/how-to-ssh-into-my-asustor-nas
> and yeah NAS are less flexible, i feel ya, same on synology.

Thank you - going to give this a try - and to answer your other questions:

1. no - did not create a stack - I'm messing with Docker / Portainer for the first time - primarily driven by HD Sentinel - which is what is the driver to learn this stuff 

My order of sequence on my NAS is:
(a) install docker-ce
(b) install portainer.io (due to the lack of a console (thanks for the SSH tip)
(c) pull your docker image via portainer.io
(d) add to container 

2.  Yes - I did set privileged mode - which is what makes this more amusing - given that it ran ONCE and I saw HD Sentinel identifying HD status, but I didn't note what parameters I was using - so I know I was close

3. Host path makes sense, I noted that I was using "Volumes" instead of "Bind" so I tried the Bind method, interestingly, Portainer created a folder named "hdsentinel" within my shared folder "HDSentinel", so something IS right, I suspect it's a privileged issue.

Going to try via SSH next - thanks for all your time and help, I appreciate the guidance.

Cheers,
Alex

**Comment by Alex000Lexx** [comment]
Hey Scyto, 

Thanks for your advice, interestingly, it has NO issues running as a command via SSH, so I am going to continue looking to see what else I might be missing. 

One lens is using cron to see if I can have it run as a scheduled task periodically within the dir itself, and then I am halfway there.

I'll keep messing around with docker-ce and portainer.io, but thanks for the guidance!

Cheers,
Alex

![HD Sentinel - working via SSH](https://user-images.githubusercontent.com/62408651/77155476-f1e01800-6a73-11ea-9123-961c11fd6c3f.jpg)
![HD Sentinel - working via SSH - 2](https://user-images.githubusercontent.com/62408651/77155487-f73d6280-6a73-11ea-8848-f1c4a604209f.jpg)

**Comment by scyto** [comment]
note on volume vs bind - get this when you -v on the command line, that is a bind!  a volume in moby speak is a private volume not a mapped folder from the host.  why docker messed terminology around like this is beyond me :-(

good luck and keep playing, yes i did all of this like you to learn, you will get there, read the docs for docker and portainer and the portainer forum is great!  good luck.  If i have time i may try and create a compose for this - i need to start doing that in addition to using command line:-)

**Comment by Alex000Lexx** [comment]
Thanks scyto! I appreciate it - what is interesting is that I have HD Sentinel working perfectly as a cron job now, that I natively configured to run directly on the NAS, via SSH, as per your great advice. I don't see any ill effects on the NAS thus far, but I wouldn't have gotten there without your advice. 

A curiosity is, when I can run a program such as hd sentinel natively on the NAS, is there a good rationale for doing it via Docker / Portainer? I can see for other programs, like a web server or database or torrents, if I want compartmentalize them, that this approach makes sense. I guess portability, ease of install / remove are the key points. 

Re: your comments on the volume vs. bind, that is great! see, I had NO clue about that nuance, but appreciate your perspective... I am going to eventually get to the point where I have this same setup working via docker, but I'll admit, it's still a concept that I am wrapping my head around (using docker containers), but I'll get there!

Thanks again for your help!  

Cheers,
Alex

**Comment by scyto** [comment]
You are right running it natively or in docker doesn't do much from a security or functionality perspective.

I did it on the synology because their support folks don't like arbitrary files being executed as root natively so this hides it from them, lol.  It also means if the file system is purged as part of an update that i can easily re-created.  also it means if / when i update the container it is easy for folks to update by doing a new pull.

so this was fun learning exercise for me in doing a docker container in github and publishing to docker hub :-)

tl;dr running natively is fine!

**Comment by Alex000Lexx** [comment]
> You are right running it natively or in docker doesn't do much from a security or functionality perspective.
> 
> I did it on the synology because their support folks don't like arbitrary files being executed as root natively so this hides it from them, lol. It also means if the file system is purged as part of an update that i can easily re-created. also it means if / when i update the container it is easy for folks to update by doing a new pull.
> 
> so this was fun learning exercise for me in doing a docker container in github and publishing to docker hub :-)
> 
> tl;dr running natively is fine!

Ahh, I see, so that is a key difference between an ASUSTor NAS and a Synology, your rationale makes perfect sense to me! If it wasn't for your efforts, I potentially might have not gone down this path, so it's great that you keep making this available to the HD Sentinel community. 

As a note, HD Sentinel is up to v.18.xxx I think you are still packaging v.17, so that might be something to consider. 

Ya, I like this approach, I am going to keep messing with docker and portainer to see what else there is that I can experiment with on my NAS.

Cheers,
Alex
